### ISSUE TITLE
Feature request: Import / Export from / to CSV, JSON

### ISSUE THREAD

**Issue by nitrocode** [phpservermon/phpservermon#250]
Would make it easier if we could import / export to or from a csv or json

**Comment by brumle80** [comment]
I concur that, easier to add large amount of servers

**Comment by mooleshacat** [comment]
Yes I duplicated issue https://github.com/phpservermon/phpservermon/issues/677 my bad

**Comment by TimZ99** [comment]
Great idea. What data do you want to see exported? Exporting all of the updates would make for a pretty big file.

**Comment by mooleshacat** [comment]
@TimZ99 it would be nice to have backup/restore where it just gives a .tar.gz file

As for import/export options, export to CSV for example, just all the settings data or notification/email logs (not the update logs)

> server 1, mdomain.com,80,service,1,true,true
> server 2, mdomain2.com,80,service,1,true,true
> server 3, mdomain3.com,80,service,1,true,true

and

> Mogwaicoin (45.77.205.34:20000),Server 'Mogwaicoin' is RUNNING: ip=45.77.205.34, port=20000, it was down for 1 minute,11/21/18 05:07:01
> Rapturecoin (45.77.205.34:20020),Server 'Rapturecoin' is DOWN: ip=45.77.205.34, port=20020, it was down for 1 minute,11/21/18 05:07:01
> Securetag (45.77.205.34:20030),Server 'Securetag' is DOWN: ip=45.77.205.34, port=20030, it was down for 1 minute,11/21/18 05:07:01

For above to work you would have to change delimiter, or remove all ',' from the error messages.

Or you could read the log data, then parse/export it as such:
> Mogwaicoin,45.77.205.34,20000,RUNNING,down for 1 minute,11/21/18 05:07:01
> Ravencoin,45.77.205.34,20020,RUNNING,down for 9 minute,11/21/18 05:07:01
> Securetag,45.77.205.34,20030,RUNNING,down for 5 minute,11/21/18 05:07:01

**[EDITED]**

**Comment by nitrocode** [comment]
@TimZ99 @leshacat yes I like this format because that it something techops can easily configure in a google spreadsheet and we could automate downloading the spreadsheet as a csv and importing into phpservermon

**Comment by mooleshacat** [comment]
We need more brains for more input here LOl ":P

**Comment by gmoulard** [comment]
I verry interesting by export prob data, not only uptime, but also Latency.
G.

**Comment by Yacine-krk** [comment]
Hi
That would be indeed a good feature to have.
I have hundreds of websites to monitor and adding them manually is painful :)

**Comment by wienfuchs** [comment]
Referencing import:
As far as I see, the import may not be too difficult, as the relating "psm_servers" table has already easy related colums. Maybe you simply want to have a look if import could not be done in phpmyadmin or so, filling out these columns.
![grafik](https://user-images.githubusercontent.com/44177781/80867430-3a563c80-8c94-11ea-8994-eca3fc5e8302.png)

Mass-adding users related to the servers may get much more difficult, as those are represented by a single table reating only server id to user id  ... but if there is only one single user, a simple sql statement may help on this as well - it creates a entry for every server for a single user with 1 in this example (it ignores possible exiting entries so only adds new created ones):
`INSERT IGNORE INTO psm_users_servers SELECT 1,server_id FROM psm_servers`

**Comment by Yacine-krk** [comment]
Thanks a lot for you answer.
That's already what I did.  And even for multiple users, as the ID is incremental, it can be done with SQL.
